### PR TITLE
NAS-116678 / 22.12 / Refuse to download update if insufficient space avail

### PR DESCRIPTION
**Pull request by anodos325**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/update_/download.py [hunk 1]
```diff
@@ -38,6 +39,13 @@ def download_impl_scale(self, job, train, location, progress_proportion):
                 self.middleware.logger.warning("Invalid update file checksum %r, re-downloading", checksum)
                 os.unlink(dst)
 
+            st = os.statvfs(os.path.dirname(dst))
+            bsize = st.f_bsize if st.f_bsize == 4096 else 512
+            avail = st.f_bavail * bsize
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/update_/download.py:44]: `filesystem.statfs` counts this as `'avail_bytes': st.f_bavail * st.f_frsize,`, why is it different here?

File: src/middlewared/middlewared/plugins/update_/download.py [hunk 2]
```diff
@@ -38,6 +39,13 @@ def download_impl_scale(self, job, train, location, progress_proportion):
                 self.middleware.logger.warning("Invalid update file checksum %r, re-downloading", checksum)
                 os.unlink(dst)
 
+            st = os.statvfs(os.path.dirname(dst))
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/update_/download.py:42]: `os.path.dirname(dst) == location`

File: src/middlewared/middlewared/plugins/update_/download.py [hunk 3]
```diff
@@ -38,6 +39,13 @@ def download_impl_scale(self, job, train, location, progress_proportion):
...
+            bsize = st.f_bsize if st.f_bsize == 4096 else 512
+            avail = st.f_bavail * bsize
+            if (1024**3) * 2 > avail:
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/update_/download.py:45]: Let's add `size` to https://update.freenas.org/scale/TrueNAS-SCALE-Angelfish/manifest.json (it is built by scale-build repo)

**Review by themylogin — CHANGES_REQUESTED** [review]

**Comment by anodos325** [comment]
This should not be merged until we get a successful build with the manifest changes and validate that it works correctly.

**Review by yocalebo — APPROVED** [review]

**Pull request merged**